tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/ManagedIdentityTests.NetFwk.cs

```diff
@@ -184,6 +189,90 @@ public async Task AcquireMsiToken_ForTokenExchangeResource_Successfully()
+        public async Task AcquireMsiToken_ExchangeForEstsToken_Successfully()
+        {
+            string resource = "api://AzureAdTokenExchange";
+
+            //Arrange
+            using (new EnvVariableContext())
+            {
+                // Fetch the env variables from the resource and set them locally
+                Dictionary<string, string> envVariables =
+                    await GetEnvironmentVariablesAsync(MsiAzureResource.WebApp).ConfigureAwait(false);
+
+                //Set the Environment Variables
+                SetEnvironmentVariables(envVariables);
+
+                //Reset cached source with update in environment variables
+                ManagedIdentityClient.resetCachedSource();
+
+                //form the http proxy URI 
+                string uri = s_baseURL + $"MSIToken?" +
+                    $"azureresource={MsiAzureResource.WebApp}&uri=";
+
+                //Create CCA with Proxy
+                IManagedIdentityApplication mia = CreateMIAWithProxy(uri, LabVaultAccessUserAssignedClientID, UserAssignedIdentityId.ClientId);
+
+                AuthenticationResult result;
+                //Act
+                result = await mia
+                            .AcquireTokenForManagedIdentity(resource)
+                            .ExecuteAsync().ConfigureAwait(false);
+
+                //Assert
+                //1. Token Type
+                Assert.AreEqual("Bearer", result.TokenType);
+
+                //2. First token response is from the MSI Endpoint
+                Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+                //3. Validate the ExpiresOn falls within a 24 hour range from now
+                CoreAssert.IsWithinRange(
+                                DateTimeOffset.UtcNow + TimeSpan.FromHours(0),
+                                result.ExpiresOn,
+                                TimeSpan.FromHours(24));
+
+                result = await mia
+                    .AcquireTokenForManagedIdentity(resource)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                //4. Validate the scope
+                Assert.IsTrue(result.Scopes.All(resource.Contains));
+
+                //5. Validate the second call to token endpoint gets returned from the cache
+                Assert.AreEqual(TokenSource.Cache,
+                    result.AuthenticationResultMetadata.TokenSource);
+
+                //6. Gets a token for the user-assigned Managed Identity.
+                var miAssertionProvider = async (AssertionRequestOptions _) =>
+                {
+                    var miResult = await mia.AcquireTokenForManagedIdentity(resource)
+                        .ExecuteAsync()
+                        .ConfigureAwait(false);
+
+                    return miResult.AccessToken;
+                };
+
+                //7. Get a token for the ESTS resource using MI token as an assertion
```

Comment:
I would add: 

- step 8: get token from ESTS again - check that it is cached (reuse the app obj)
- step 9: get token from ESTS and use force refresh - check that the MI provider returns a cached token for api://tokenExchange